src/import_edf.jl

```diff
@@ -208,81 +302,264 @@ function groupby(f, list)
     return d
 end
 
-"""
-    extract_channels_by_label(edf::EDF.File, signal_names, channel_names;
-                              unit_alternatives=STANDARD_UNITS, preprocess_labels=(l,t) -> l)
+# unpack a single channel spec from labels:
+# "channel"
+canonical_channel_name(channel_name) = channel_name
+# "channel" => ["alt1", "alt2", ...]
+canonical_channel_name(channel_alternates::Pair) = first(channel_alternates)
 
-For one or more signal names and one or more channel names,
-return a list of `[(infos, errors)...]` where `errors` is a vector of errors that occurred,
-and `infos` is a vector of `(si::Onda.SamplesInfo, edf_signals::Vector{EDF.Signal})`,
-where `edf_signals` align with `si.channel`. This list can have more than one pair
-if channels with the same signal kind/type have different sample rates or
-physical units.
+plan_edf_to_onda_samples(signal::EDF.Signal, s; kwargs...) = plan_edf_to_onda_samples(_named_tuple(signal.header), s; kwargs...)
+plan_edf_to_onda_samples(header::EDF.SignalHeader, s; kwargs...) = plan_edf_to_onda_samples(_named_tuple(header), s; kwargs...)
 
-`(label, transducer_type)` pairs will be transformed into labels by `preprocess_labels`
-(default preprocessor returns the original label). An alternate to `STANDARD_UNITS`
-for mapping different spellings of units to their canonical unit name can be passed in
-as `unit_alternatives`.
+"""
+    plan_edf_to_onda_samples(header, seconds_per_record; labels=STANDARD_LABELS,
+                             units=STANDARD_UNITS, preprocess_labels=(l,t) -> l)
```

Comment:
Looks like the purpose of the `preprocess_labels` keyword argument isn't documented. I don't know what I expected but I did not expect the arguments passed to it to be the label and the transducer type as I've never seen the transducer type used for anything meaningful before. Since `preprocess_labels` is a function of `header` only, does it have to be the job of `plan_edf_to_onda_samples` to call it?

Reply:
This has been very useful in some data we've seen "in the wild" where the label and transducer were basically flipped in some recordings.  In practice it was much more effective to do this kind of preprocessing rather than try to mutate the EDF file headers.

I do wonder whether, in this brave new world, that kind of situation would be better handled by working directly with the `plan_edf_to_onda_samples(signal::EDF.Signal)` level, since we can :just: manipulate the named tuple for the signal header.  There's some disadvantage to that, sicne the `EDF.File` function does a fair bit of work to sort/group things, which isn't factored out into a separate function that you could apply to a vector of plans.  But maybe this is a good reason to do that kind of refactoring?

Reply:
If we accept a signal header or a corresponding `NamedTuple` then I think it's reasonable for callers to do their own custom preprocessing and pass the result to this function rather than handling the calling for them. Now _that_'s inverting control.

Reply:
Yeah, that's fair.  The issue is that it's a bit clunky then to do the file-level operations (like assigning `edf_signal_index` and `onda_signal_index`), since that's all handled by the `plan_edf_to_onda_samples(::EDF.File, ...)` method which just forwards it's kwargs to the `plan_edf_to_onda_samples(header, ...)` method.